.github/workflows/pr_check.yml

```diff
@@ -0,0 +1,26 @@
+name: PR Check
+
+on:
+  pull_request:
+    types: [opened, synchronize, reopened, edited]
+    branches:
+      - main
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref == 'refs/heads/main' && github.run_id || github.event.pull_request.number || github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  pr_title_check:
+    runs-on: ubuntu-22.04
+    name: Title format
+    steps:
+      - uses: actions/checkout@v4
+      - name: Bootstrap
+        uses: ./.github/actions/bootstrap
```

Comment:
if we only need Python and pip, we could also use the official Python action: https://github.com/actions/setup-python.

Reply:
I used the bootstrap action to make sure that if we need to update the Python version, it is all in the same place